corehq/apps/userreports/sql/adapter.py

```diff
@@ -208,6 +208,9 @@ def bulk_save(self, docs):
         self.save_rows(rows)
 
     def bulk_delete(self, docs, use_shard_col=True):
+        if not self.table_exists:
```

Comment:
Does this fire a new query or it already is aware?

Reply:
Good question. 

The `table_exists` method invokes the `has_table` method on the engine object. The engine is [created here](https://github.com/dimagi/commcare-hq/blob/a3ba7433eee3e6bfc6604ed4726c994978e1a170/corehq/sql_db/connections.py#L35-L40). Looking manually through the sqlalchemy code it looks like it uses cached info. 

I'm just wondering now, if it's using cached info, what's the chance of it getting out of date? I'll try and see if I can find out more about how this works.

Reply:
@mkangia Ok, I can't find good documentation on how the caching updates work, so the safest course is probably to just catch the exception, since we're making a DB request in any case.

Reply:
Here's a [draft PR](https://github.com/dimagi/commcare-hq/pull/34595)